Source/WebCore/css/parser/CSSParserSelector.cpp

```diff
@@ -50,8 +50,12 @@ std::unique_ptr<CSSParserSelector> CSSParserSelector::parsePagePseudoSelector(St
     return selector;
 }
 
-std::unique_ptr<CSSParserSelector> CSSParserSelector::parsePseudoElementSelector(StringView pseudoTypeString)
+std::unique_ptr<CSSParserSelector> CSSParserSelector::parsePseudoElementSelector(StringView pseudoTypeString, CSSParserMode mode)
 {
+    if (mode != UASheetMode) {
+        if (equalLettersIgnoringASCIICase(pseudoTypeString, "thumb"_s) || equalLettersIgnoringASCIICase(pseudoTypeString, "track"_s))
+            return nullptr;
+    }
```

Comment:
This logic is likely going to run too frequently. After looking at this code more closely, my suggestion would be to:

1. Remove the changes here.
2. Give your pseudo element a `-webkit` prefix.
3. Update `CSSSelector::parsePseudoElementType` to return `PseudoElementUnknown` in the exceptional case, underneath `type == PseudoElementWebKitCustom`.

Then the logic wouldn't run for any non-custom pseudo elements.

Reply:
I disagree we should add a `-webkit-` prefix. The end goal is to eventually expose this pseudo to the web. We should properly support shadow pseudo IDs without `-webkit-` prefixes.

Reply:
Sounds good. We should still restrict the additional check behind a comparison to `PseudoElementWebKitCustom` though, to avoid unnecessary comparisons.